components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.ts

```diff
@@ -0,0 +1,335 @@
+        this.overrideTab = true;
+        break;
+    }
+  }
+
+
+  // Handles the data of cookbook version array coming from constraint tab.
+  constraintItemsHandler(value: Array<DynamicGrid> = []    ) {
+    this.constraintArray = value;
+  }
+
+  // Getting list of cookbook names
+  loadCookbookConstraint() {
+    this.name_id = '';
+    this.store.dispatch(new GetCookbooks({
+      server_id: this.serverId, org_id: this.orgId
+    }));
+
+    combineLatest([
+      this.store.select(getAllCookbooksForOrgStatus),
+      this.store.select(allCookbooks)
+    ]).pipe(takeUntil(this.isDestroyed))
+    .subscribe(([ getCookbooksSt, allCookbooksState]) => {
+      if (getCookbooksSt === EntityStatus.loadingSuccess && !isNil(allCookbooksState)) {
+        this.constraintKeys = [];
+        this.cookbooks = allCookbooksState;
+        this.cookbooks.forEach((cookbook) => {
+          this.constraintKeys.push(cookbook.name);
+        });
+      }
+      this.name_id = this.constraintKeys[0];
+    });
+
+  }
+
+  createEnvironment() {
+    this.creating = true;
+    const environment = {
+      org_id: this.orgId,
+      server_id: this.serverId,
+      name: this.detailsFormGroup.controls['name'].value,
+      description: this.detailsFormGroup.controls['description'].value,
+      cookbook_versions: this.constraintArray.length ? this.toDisplay(this.constraintArray) : {},
+      default_attributes: JSON.parse(this.defaultAttrFormGroup.controls['default'].value),
+      override_attributes: JSON.parse(this.overrideAttrFormGroup.controls['override'].value)
+    };
+
+    this.store.dispatch(
+      new CreateEnvironment({environment: environment})
+    );
+  }
+
+  toDisplay(cookbookVersions: Array<DynamicGrid> = []) {
+    const current = {};
+    cookbookVersions.forEach((element) => {
+      current[element.name] =
+        `${element.operator}` + ' ' + `${element.version}`;
+    });
+    return current;
+  }
+
+  closeCreateModal(): void {
+    this.resetCreateModal();
+    this.visible = false;
+  }
+
+  private resetCreateModal(): void {
+    this.creating = false;
+    this.detailsFormGroup.reset();
+    this.defaultAttrFormGroup.reset();
+    this.overrideAttrFormGroup.reset();
+    this.showConstraint = false;
+    this.overrideTab = false;
+    this.defaultTab = false;
+    this.constraintsTab = false;
+    this.detailsTab = true;
+    this.defaultAttrFormGroup.controls.default.setValue(this.attr_value);
+    this.overrideAttrFormGroup.controls.override.setValue(this.attr_value);
+    this.defaultAttrParseError = false;
+    this.overrideAttrParseError = false;
+    this.dynamicArray = [];
+    this.loadCookbookConstraint();
+    this.conflictErrorEvent.emit(false);
+  }
+
+  private resetTabs() {
+    this.detailsTab = false;
+    this.constraintsTab = false;
+    this.defaultTab = false;
+    this.overrideTab = false;
+  }
+
+  ngOnDestroy(): void {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
```

Comment:
Suggest move this to be right after `ngOnInit`.

Reply:
added